### PR TITLE
Cleanup and golf a few proofs

### PR DESCRIPTION
**Pull request by pitmonticone**

### CODE REVIEW

File: PFR/Main.lean [hunk 1]
```diff
@@ -174,7 +174,8 @@ lemma PFR_conjecture_aux (h₀A : A.Nonempty) (hA : Nat.card (A + A) ≤ K * Nat
     PFR_conjecture_pos_aux h₀A hA
   let A' := A.toFinite.toFinset
   have h₀A' : Finset.Nonempty A' := by
-    simp [A', Finset.Nonempty]
+    simp only [Finset.Nonempty, Finite.toFinset_setOf, Finset.mem_filter, Finset.mem_univ, true_and,
+      A']
     exact h₀A
```
Comment (YaelDillies) [inline PFR/Main.lean:179]:
```suggestion
    simp [A', Finset.Nonempty] using h₀A
```
is more robust to lemma renames

File: PFR/WeakPFR.lean [hunk 2]
```diff
@@ -1240,7 +1235,7 @@ max (dimension A') (dimension A'') := by
     convert zpow_neg K 17 using 1
     norm_cast
   calc (dimension B : ℝ)
-    _ ≤ (((max (dimension A') (dimension A'')) : ℕ) : ℝ) := by norm_cast
+    _ ≤ (((max (dimension A') (dimension A'')) : ℕ) : ℝ) := Nat.cast_le.mpr hBdim
```
Comment (YaelDillies) [inline PFR/WeakPFR.lean:1238]: That's less robust than what was before

File: PFR/Main.lean (hunk 1 above)
Comment (pitmonticone) [reply PFR/Main.lean:179]: I didn't think about that. Thank you very much! I'll take robustness to renaming in the future! 

I'll commit your suggestion later today when I'll be back home.

File: PFR/WeakPFR.lean (hunk 2 above)
Comment (pitmonticone) [reply PFR/WeakPFR.lean:1238]: Again, thanks for the explanation. I'll do it later.

File: PFR/Main.lean (hunk 1 above)
Comment (pitmonticone) [reply PFR/Main.lean]: Do you mean `simpa [A', Finset.Nonempty] using h₀A`, right? 

Done.

File: PFR/WeakPFR.lean (hunk 2 above)
Comment (pitmonticone) [reply PFR/WeakPFR.lean]: Done. Thanks!

File: PFR/Main.lean (hunk 1 above)
Comment (YaelDillies) [reply PFR/Main.lean]: Yes, sorry! Hard to review from a phone 😅

**Pull request merged**